core/storage/wal.rs

```diff
@@ -264,16 +277,33 @@ impl WalFile {
                         self.io.run_once()?;
                         self.wal_header.replace(Some(wal_header));
                     } else {
-                        let wal_header = WalHeader {
-                            magic: (0x377f0682_u32).to_be_bytes(),
+                        // magic is a single number represented as WAL_MAGIC_LE but the big endian
+                        // counterpart is just the same number with LSB set to 1.
+                        let magic = WAL_MAGIC_LE | cfg!(target_endian = "big") as u32;
```

Comment:
just my grug brain, but imo this would read simpler as 
```
let magic = if cfg!(target_endian = "big") {
  WAL_MAGIC_BE
} else {
  WAL_MAGIC_LE
}
```

Reply:
the comment could be moved into the definitions of WAL_MAGIC_*